### ISSUE TITLE
torch.gels on GPU doesn't support m < n

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#3205]
[The documentation for torch.gels](http://pytorch.org/docs/master/torch.html#torch.gels) suggests that when m < n for the input matrix A, then it solves the least norms problem. torch.gels with cuda tensor arguments doesn't do this.
```
import torch
b = torch.randn(3 ,1)
A = torch.randn(3, 5)
X, _ = torch.gels(b,A)
Y, _ = torch.gels(b.cuda(),A.cuda())  
>> Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: invalid argument 2: A should have m >= n at /home/rzou/pytorch/torch/lib/THC/generic/THCTens
orMathMagma.cu:107
```

**Comment by vishwakftw** [comment]
I feel that this is a documentation issue, in which case #23460 should fix this.

**Comment by colesbury** [comment]
@vishwakftw I don't see how renaming this to lstsq fixes this. The NumPy docs say "The equation may be under-, well-, or over-determined...". Doesn't under-determined correspond to M >= N?

**Comment by vishwakftw** [comment]
@colesbury We don't have support for an under-determined system (M < N) on CUDA, as neither MAGMA nor cuBLAS provide support for the same.

In the renaming PR, I ensured that this was clarified in the documentation by stating that m < n is not supported, hence the reference.
https://github.com/pytorch/pytorch/blob/b67ee90b854af25f0f1f768a2807c2d2632da1ee/torch/_torch_docs.py#L2632-L2633

**Comment by colesbury** [comment]
@vishwakftw sounds good. It would be nice to improve the error message as well.

**Comment by vishwakftw** [comment]
@colesbury The error has already been revised in https://github.com/pytorch/pytorch/commit/bacda6df8dada71850cc228ddd12530ba7434f72.

**Comment by vishwakftw** [comment]
Closed by #23460
